jax/lib/xla_bridge.py

```diff
@@ -59,36 +59,42 @@
     'pass "cpu" for CPU or "gpu" for GPU.')
 
 
-def get_compile_options(num_replicas=None, device_assignment=None):
+def get_compile_options(num_replicas, num_partitions, device_assignment=None):
   """Returns the compile options to use, as derived from flag values.
 
   Args:
-    num_replicas: Optional int indicating the number of replicas for which to
-      compile (default inherited from xla_client.CompileOptions).
+    num_replicas: int indicating the number of replicas for which to compile.
+    num_partitions: int indicating the number of partitions for which to compile.
     device_assignment: Optional tuple of integers indicating the assignment of
       logical replicas to physical devices (default inherited from
-      xla_client.CompileOptions). Must be consistent with `num_replicas`.
+      xla_client.CompileOptions). Must be consistent with `num_replicas` and
+      `num_partitions`.
   """
-  compile_options = None
-  if num_replicas is not None:
-    compile_options = compile_options or xla_client.CompileOptions()
-    compile_options.num_replicas = num_replicas
+  compile_options = xla_client.CompileOptions()
+  compile_options.num_replicas = num_replicas
+  compile_options.num_partitions = num_partitions
```

Comment:
Maybe you already figured this out, but this isn't actually doing anything until we build a new jaxlib that includes https://github.com/tensorflow/tensorflow/commit/d479ba4e1dd7edec404a2cd84c69ccafee866709 (which I'm in the process of building now). I guess this works as-is because it will create a `num_partitions` field on the fly, and we don't depend on the new jaxlib behavior anywhere.

That said, I think we should bump the [minimum jaxlib version](https://github.com/google/jax/blob/master/jax/lib/__init__.py#L20) to 0.1.38 as part of this change (and I can merge it tomorrow after I upload 0.1.38 jaxlib wheels). Even though this technically passes tests, it doesn't implement the expected behavior if num_partitions > 1. WDYT?

Reply:
Yes, that's true. It can wait until tomorrow.

Reply:
Ok, we have new wheels and Peter already bumped the minimum version in https://github.com/google/jax/commit/991324f8dfa7fea3b985be5e99b130db29409074. I'll go ahead and merge this.